presto-main/src/main/java/com/facebook/presto/sql/planner/plan/MergeJoinNode.java

```diff
@@ -47,20 +50,24 @@
     public MergeJoinNode(
             Optional<SourceLocation> sourceLocation,
             @JsonProperty ("id") PlanNodeId id,
-            @JsonProperty("type") JoinNode.Type type,
-            @JsonProperty("left") PlanNode left,
-            @JsonProperty("right") PlanNode right,
-            @JsonProperty("criteria") List<JoinNode.EquiJoinClause> criteria,
-            @JsonProperty("outputVariables") List<VariableReferenceExpression> outputVariables,
-            @JsonProperty("filter") Optional<RowExpression> filter,
-            @JsonProperty("leftHashVariable") Optional<VariableReferenceExpression> leftHashVariable,
-            @JsonProperty("rightHashVariable") Optional<VariableReferenceExpression> rightHashVariable)
+            @JsonProperty ("type") JoinNode.Type type,
+            @JsonProperty ("left") PlanNode left,
+            @JsonProperty ("right") PlanNode right,
+            @JsonProperty ("criteria") List<JoinNode.EquiJoinClause> criteria,
+            @JsonProperty ("leftSortingProperties") List<SortingProperty<VariableReferenceExpression>> leftSortingProperties,
```

Comment:
As discussed offline, let's separate this into two PRs:
One PR to fix the correctness issues. Make sure we only plan merge join for the supported use cases. Disable the merge join planning for cases like the join column order doesn't match sort column order and the cases with DESC orders. This PR should not introduce any new fields so it doesn't require CPP side changes. This can unblock the CPP benchmark.

Another PR to support more use cases as mentioned above by adding these additional fields. These are nice-to-have features(low priority) and are not blockers for the benchmark. It requires CPP side change to update the protocols.